pkg/uio/buffer.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uio
+
+import (
+	"encoding/binary"
+	"io"
+)
+
+// Marshaler is the interface implemented by an object that can marshal itself
+// into binary form.
+//
+// Marshal appends data to the buffer b.
+type Marshaler interface {
+	Marshal(b *Buffer)
+}
+
+// Unmarshaler is the interface implemented by an object that can unmarshal a
+// binary representation of itself.
+//
+// Unmarshal consumes data from the buffer b.
+type Unmarshaler interface {
+	Unmarshal(b *Buffer) error
+}
+
+// Buffer implements functions to manipulate byte slice buffers in specific
+// byte orders.
+type Buffer struct {
+	// data is the underlying data.
+	data []byte
+
+	// order is the byte order to write in / read in.
+	order binary.ByteOrder
+}
+
+// NewBEBuffer consumes b for marshaling or unmarshaling in big endian.
+func NewBEBuffer(b []byte) *Buffer {
```

Comment:
I had to read these names twice to see they were different. 
Since we have the long names in other packages already, could we just do
NewLittleEndianBuffer
NewBigEndianBuffer

and, further, is there any way to get
NewNativeEndianBuffer
because that's actually handy at times.